xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -447,15 +398,75 @@ bool CPVRClient::GetAddonProperties()
   if (retVal == PVR_ERROR_NOT_IMPLEMENTED)
     retVal = PVR_ERROR_NO_ERROR; // timer support is optional.
 
+  /* update the members */
+  std::unique_lock<CCriticalSection> lock(m_critSection);
+  m_clientCapabilities = addonCapabilities;
+  m_timertypes = timerTypes;
+
+  return retVal == PVR_ERROR_NO_ERROR;
+}
+
+bool CPVRClient::GetAddonNameStringProperties()
+{
+  char strBackendName[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  char strConnectionString[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  char strBackendVersion[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  char strBackendHostname[PVR_ADDON_NAME_STRING_LENGTH] = {};
+  std::string strFriendlyName;
+
+  /* get the name of the backend */
+  PVR_ERROR retVal = DoAddonCall(
+      __func__,
+      [&strBackendName](const AddonInstance* addon) {
+        return addon->toAddon->GetBackendName(addon, strBackendName, sizeof(strBackendName));
+      },
+      true, false);
+
+  if (retVal != PVR_ERROR_NO_ERROR)
+    return false;
+
+  /* get the connection string */
+  retVal = DoAddonCall(
+      __func__,
+      [&strConnectionString](const AddonInstance* addon) {
+        return addon->toAddon->GetConnectionString(addon, strConnectionString,
+                                                   sizeof(strConnectionString));
+      },
+      true, false);
+
+  if (retVal != PVR_ERROR_NO_ERROR && retVal != PVR_ERROR_NOT_IMPLEMENTED)
+    return false;
+
+  /* backend version number */
+  retVal = DoAddonCall(
+      __func__,
+      [&strBackendVersion](const AddonInstance* addon) {
+        return addon->toAddon->GetBackendVersion(addon, strBackendVersion,
+                                                 sizeof(strBackendVersion));
+      },
+      true, false);
+
+  if (retVal != PVR_ERROR_NO_ERROR)
+    return false;
+
+  /* backend hostname */
+  retVal = DoAddonCall(
+      __func__,
+      [&strBackendHostname](const AddonInstance* addon) {
+        return addon->toAddon->GetBackendHostname(addon, strBackendHostname,
+                                                  sizeof(strBackendHostname));
+      },
+      true, false);
+
+  if (retVal != PVR_ERROR_NO_ERROR && retVal != PVR_ERROR_NOT_IMPLEMENTED)
+    return false;
+
   /* update the members */
   std::unique_lock<CCriticalSection> lock(m_critSection);
   m_strBackendName = strBackendName;
   m_strConnectionString = strConnectionString;
-  m_strFriendlyName = strFriendlyName;
   m_strBackendVersion = strBackendVersion;
-  m_clientCapabilities = addonCapabilities;
   m_strBackendHostname = strBackendHostname;
-  m_timertypes = timerTypes;
 
   return retVal == PVR_ERROR_NO_ERROR;
```

Comment:
This should read return true;. Some of the API methods (e.g. GetBackendHostname) are optional and can return PVR_ERROR_NOT_IMPLEMENTED. If that happens retVal here will contain PVR_ERROR_NOT_IMPLEMENTED which is not an error, but you would return false and the add-on will not work. Happens for example for pvr.plutotv.

Reply:
Right you are.  The AddonProperties logic is so odd if false is every returned the addon is disabled and cannot be automatically enabled.